doc/api/quic.md

```diff
@@ -0,0 +1,2276 @@
+# QUIC
+
+<!--introduced_in=REPLACEME-->
+
+> Stability: 1 - Experimental
+
+The `net` module provides an implementation of the QUIC protocol. To
+access it:
+
+```js
+const { createQuicSocket } = require('net');
+```
+
+## Example
+
+```js
+'use strict';
+
+const key = getTLSKeySomehow();
+const cert = getTLSCertSomehow();
+
+const { createQuicSocket } = require('net');
+
+// Create the QUIC UDP IPv4 socket bound to local IP port 1234
+const socket = createQuicSocket({ endpoint: { port: 1234 } });
+
+// Tell the socket to operate as a server using the given
+// key and certificate to secure new connections, using
+// the fictional 'hello' application protocol.
+socket.listen({ key, cert, alpn: 'hello' });
+
+socket.on('session', (session) => {
+  // A new server side session has been created!
+
+  session.on('secure', () => {
```

Comment:
Yes. QUIC supports TLS 1.3 0-RTT and 0.5-RTT options though neither is enabled just yet due to some timing issues that I've still to work out. 

0-RTT on the client side will mean that the client will be able to create and write data to a `QuicStream` before the TLS handshake is started. 

0.5RTT will mean that either the client or server will be permitted to create and write data to a `QuicStream` before the TLS handshake is completed.

There are also a number of events (such as `keylog` and `OCSPRequest`) that need to be applied to the `QuicSession` object before the completion of the TLS handshake. because they interact with the handshake itself.

Reply:
I assumed from the name that this was just a re-emit of https://nodejs.org/api/tls.html#tls_event_secureconnect, but since it models a different concept, perhaps it should be called a more distinct name, like `'ready'` (not used anywhere else AFAIK).